src/main/java/de/tum/in/www1/artemis/config/migration/DatabaseMigration.java

```diff
@@ -0,0 +1,256 @@
+     * (as mentioned in the thrown error message) before proceeding.
+     * <p>
+     * This method ensures that the application's database schema is compatible with the application's current version,
+     * adhering to the migration path requirements.
+     *
+     * @return The latest version string from the 'artemis_version' table if it exists.
+     * @throws RuntimeException If the 'DATABASECHANGELOG' table does not exist, or if retrieving the latest version fails.
+     */
+    private String getPreviousVersionElseThrow() {
+        String error = "Cannot start Artemis because version table does not exist, but a migration path is necessary! Please start the release 5.12.9 first, otherwise the migration will fail";
+        try (var statement = createStatement()) {
+            statement.executeQuery("SELECT * FROM DATABASECHANGELOG;");
+            var result = statement.executeQuery("SELECT latest_version FROM artemis_version;");
+            statement.closeOnCompletion();
+            if (result.next()) {
+                return result.getString("latest_version");
+            }
+            // if no version exists, we fail here
+            log.error(error);
+            throw new RuntimeException(error);
+        }
+        catch (SQLException e) {
+            if (e.getMessage().contains("databasechangelog") && (e.getMessage().contains("does not exist") || (e.getMessage().contains("doesn't exist")))) {
+                return null;
+            }
+            log.error(error);
+            throw new RuntimeException(error, e);
+        }
+    }
+
+    /**
+     * Updates the checksum of the initial schema in the 'DATABASECHANGELOG' table to {@code null}.
+     * This operation is crucial for allowing Liquibase to recalculate the checksum during the
+     * migration process, ensuring that the database schema is up-to-date with the specified new version
+     * of the application. This method specifically targets the initial schema entry, preparing it for
+     * a fresh checksum calculation by Liquibase.
+     * <p>
+     * The update is performed with the intention of aligning the database schema with the new version,
+     * thereby facilitating a smooth transition and avoiding potential migration conflicts.
+     * <p>
+     * If an SQLException occurs during the update process, a RuntimeException is thrown, indicating
+     * that the checksum update operation has failed. This failure needs to be addressed to ensure
+     * the integrity and consistency of the database schema migration process.
+     *
+     * @param newVersion The new version of the application for which the initial schema checksum needs to be updated.
+     * @throws RuntimeException If updating the checksum fails due to an SQLException, encapsulating the original exception.
+     */
+    private void updateInitialChecksum(String newVersion) {
+        // SQL statement with a placeholder for the newVersion parameter
+        String updateSqlStatement = """
```

Comment:
What about the rest of the table? Shouldn't we clear everything but the Initial schema generation first?

Reply:
No, this is done within a liquibase migration